photon-core/src/main/java/org/photonvision/vision/processes/VisionModuleChangeSubscriber.java

```diff
@@ -54,28 +57,48 @@ public void onDataChangeEvent(DataChangeEvent<?> event) {
         if (event instanceof IncomingWebSocketEvent) {
             var wsEvent = (IncomingWebSocketEvent<?>) event;
 
-            // Camera index -1 means a "multicast event" (i.e. the event is received by all cameras)
+            // Camera index -1 means a "multicast event" (i.e. the event is received by all
+            // cameras)
             if (wsEvent.cameraIndex != null
                     && (wsEvent.cameraIndex == parentModule.moduleIndex || wsEvent.cameraIndex == -1)) {
                 logger.trace("Got PSC event - propName: " + wsEvent.propertyName);
+                synchronized (m_mutex) {
+                    getSettingChanges()
+                            .add(
+                                    new VisionModuleChange(
+                                            wsEvent.propertyName,
+                                            wsEvent.data,
+                                            parentModule.pipelineManager.getCurrentPipeline().getSettings(),
+                                            wsEvent.originContext));
+                }
+            }
+        }
+    }
 
-                var propName = wsEvent.propertyName;
-                var newPropValue = wsEvent.data;
-                var currentSettings = parentModule.pipelineManager.getCurrentPipeline().getSettings();
+    public List<VisionModuleChange<?>> getSettingChanges() {
+        return settingChanges;
+    }
 
-                // special case for non-PipelineSetting changes
+    public void processSettingChanges() {
+        // special case for non-PipelineSetting changes
+        synchronized (m_mutex) {
```

Comment:
Same not about [ReentrantLock](http://docs.oracle.com/javase/8/docs/api/java/util/concurrent/locks/ReentrantLock.html)?